piksi_tools/console/sbp_relay_view.py

```diff
@@ -100,10 +148,129 @@ def update_msgs(self):
     else:
       raise NotImplementedError
 
-  def _start_fired(self):
+  def set_route(self, device_uid):
+    self.device_uid = device_uid
+    if self.http:
+      self.http.device_uid = device_uid
+
+  def _prompt_networking_error(self, text):
+    """Nonblocking prompt for a networking error.
+
+    Parameters
+    ----------
+    text : str
+      Helpful error message for the user
+
     """
-    Handle start udp broadcast button. Registers callbacks on self.link for each of the self.msgs
-    If self.msgs is None, it registers one generic callback for all messages
+    prompt = CallbackPrompt(title="Networking Error", actions=[close_button])
+    prompt.text = text
+    prompt.run(block=False)
+
+  def _prompt_setting_error(self, text):
+    """Nonblocking prompt for a device setting error.
+
+    Parameters
+    ----------
+    text : str
+      Helpful error message for the user
+
+    """
+    prompt = CallbackPrompt(title="Setting Error", actions=[close_button])
+    prompt.text = text
+    prompt.run(block=False)
+
+  def _prompt_wait(self, text, interval=5.):
+    """Configurable blocking prompt.
+
+    Parameters
+    ----------
+    text : str
+      Helpful wait message for the user.
+    interval : float
+      Time interval for waiting.
+
+    """
+    prompt = CallbackPrompt(title="waiting...", actions=[])
+    prompt.text = text
+    t0 = time.time()
+    while not self.http.connect_read() or (time.time() - t0) < interval:
+      prompt.run(block=True)
```

Comment:
i was wondering if this might have caused the console blocking issues you were seeing earlier? seems unlikely / i didn't dig into this too much.

p.s. I saw nothing.